logging/access_test.go

```diff
@@ -0,0 +1,78 @@
+package logging
+
+import (
+	"bytes"
+	"net/http"
+	"testing"
+	"time"
+)
+
+const logOutput = `127.0.0.1 - - [10/Oct/2000:13:55:36 +0000] "GET /apache_pb.gif HTTP/1.1" 418 2326 "" "" 42`
+
+func testRequest() *http.Request {
+	r, _ := http.NewRequest("GET", "http://frank@127.0.0.1", nil)
+	r.RequestURI = "/apache_pb.gif"
+	r.RemoteAddr = "127.0.0.1"
+	return r
+}
+
+func testDate() time.Time {
+	return time.Date(2000, 10, 10, 13, 55, 36, 0, time.FixedZone("Test", -7))
+}
+
+func testAccessEntry() *AccessEntry {
+	return &AccessEntry{
+		Request:      testRequest(),
+		ResponseSize: 2326,
+		StatusCode:   http.StatusTeapot,
+		RequestTime:  testDate(),
+		Duration:     42 * time.Millisecond}
+}
+
+func testAccessLog(t *testing.T, entry *AccessEntry, expectedOutput string) {
+	var buf bytes.Buffer
+	Init(Options{AccessLogOutput: &buf})
+	Access(entry)
+	got := buf.String()
+	if got != "" {
+		got = got[:len(got)-1]
+	}
+
+	if got != expectedOutput {
+		t.Error("got wrong access log.")
+		t.Log("expected:", expectedOutput)
+		t.Log("got     :", got)
+	}
+}
+
+func TestAccessLogFormatFull(t *testing.T) {
+	testAccessLog(t, testAccessEntry(), logOutput)
+}
+
+func TestAccessLogIgnoresEmptyEntry(t *testing.T) {
+	testAccessLog(t, nil, "")
+}
+
+func TestNoPanicOnMissingRequest(t *testing.T) {
+	entry := testAccessEntry()
+	entry.Request = nil
+	testAccessLog(t, entry, `- - - [10/Oct/2000:13:55:36 +0000] "  " 418 2326 "" "" 42`)
+}
+
+func TestUseXForwarded(t *testing.T) {
```

Comment:
Can you add a test case for the case, that `X-Forwarded-For` contains multiple IP addresses?
Something like:
```
X-Forwarded-For: 1.2.3.4, 5.4.3.2, 8.8.8.8
```